mlir/lib/Dialect/Tensor/IR/TensorOps.cpp

```diff
@@ -1622,7 +1623,20 @@ CollapseShapeOp::inferCollapsedType(RankedTensorType type,
     currentDim += dim;
   }
 
-  return RankedTensorType::get(newShape, type.getElementType());
+  auto encoding = type.getEncoding();
+  if (auto v = encoding.dyn_cast_or_null<VerifiableTensorEncoding>()) {
+    auto ignoreError = [&] {
+      auto emitter = mlir::emitError(UnknownLoc::get(type.getContext()));
+      emitter.abandon();
+      return emitter;
+    };
+    if (failed(
```

Comment:
@aartbik , this was the change that @matthias-springer wanted to call out. This was a real case i ran into where the encoding could no longer be applied to the derived type since the dimensionality changed. 

Thoughts?

Reply:
Yeah, when dimensionality changes, one would have to come up with a new encoding (e.g. a sparse matrix becomes a sparse vector or the opposite). But that indeed seems to be out of scope for this change, so reporting an error seems the right approach

Reply:
@aartbik thank you for the reply.

You say "an error seems the right approach"

I want to be clear on two things:
 1. The change i have now is *not* emitting an error; its falling-back to the pre-commit behavior of just dropping the attribute. 
 2. Changing the code to emit error in this case will break exiting code, including some tests.
 
 Given this, are you okay with the code as is? Alternatively, i can emit an error here and make the failing tests as expected fail? I'm okay with either option.

Reply:
Sorry for the late reply. Yeah, by "error" I really meant not propagating (to a sparse guy, that is always an error, but I don't know why I said that here ;-)

In principle, dropping sparsity is always "correct", i.e. we will fall back to the dense state of things. Note that we have some nifty new ideas on how to efficiently change ranks (viz. view a sparse matrix as a sparse vector), but that should really be audited very carefully.

So this is good to go!